### ISSUE TITLE
replaying on rds

### ISSUE THREAD

**Issue by rbase8888** [spaghettidba/DBA-Scripts#1]
some definitions 

v1: (EC2) aws account is a single gigantic lift and shift approach with no proper redundancy, configuration management, or orchestration

v2: (RDS) is a set of separated accounts that follow aws best practices, fully leveraging aws services as the first option starting completely with configuration as code

ok now the issues im experiencing 

workload tools is collecting xevents fine, and storing them for replay from v1 - but when I try to replay sql statements on v2 there are two issues -

1. prepared statements do not contain all the metadata in v2 to run properly and i would need to update the statement in the workload tools db to get it to run without the metadata. is there a configuration in workload tools to strip that out so prepared statements are more portable?

2. overall when workload tools replays the sessions in v2 i get quite a few locks than when they are replayed manually. is there an isolation level being set too aggressively? some other workload tools session config that needs to be dialed back related to transactions or locking? to restate, rerunning these same queries manually don't seem to have the same problems

i will be poking around after this post to determine if there is a solution and i will post if i find any thing 
best 
Rob

**Comment by spaghettidba** [comment]
Hi Rob, 

I'm not sure I understand what you mean with "metadata"? Can you give an example?
Regarding locks, WT replays all the commands, including SET TRANSACTION ISOLATION LEVEL commands. It should hence use the original isolation level. Have you investigated the isolation level being used, compared to the one you get when running manually?

Thanks
Gianluca

**Comment by rbase8888** [comment]
hey,
is there a way to set the SET TRANSACTION ISOLATION LEVEL configuration for replaying the traffic?
if that makes since lol
best

**Comment by spaghettidba** [comment]
Currently no. It is something that we could discuss implementing. How would you like it to behave?
I am not sure I can prevent the original transaction isolation level from being used (could be inside stored procedures and the like). We can at least try.

**Comment by rbase8888** [comment]
ok so we were able to get around the locking issues we were experiencing.. on our rds instance we changed our MAXDOP from "unlimited" to 1 and increased out parallelism from 50 to 100.. so if any one else is experiencing lots of weird locking that could help..  we are still having issues running prepared statements tho...  there seems to be missing metadata from when we try to replay the ec2 traffic on our rds instance.. the transaction isolation levels are the same.. any other thoughts?  
Warn - WorkloadTools.Consumer.Replay.ReplayWorker : Worker [200] - Sequence[26742921] - Error: Could not find prepared statement with handle 48479.

**Comment by spaghettidba** [comment]
Yes, I know that prepared statements are a weak spot of WorkloadTools. I am trying my best to honor prepared statement executions but some escape my efforts

**Comment by rbase8888** [comment]
ok next idea is there a way to "listen for rpc starting" 

https://social.msdn.microsoft.com/Forums/sqlserver/en-US/42a18ab3-79f1-4370-bf8f-13fec9081ff7/could-not-find-prepared-statement-with-handle-?forum=transactsql

https://stackoverflow.com/questions/9830881/sql-server-could-not-find-prepared-statement-with-handle-10-from-query-analizer
